packages/mui-material/src/PigmentGrid/PigmentGrid.tsx

```diff
@@ -0,0 +1,300 @@
+import * as React from 'react';
+import clsx from 'clsx';
+import PropTypes from 'prop-types';
+import { OverridableComponent, OverrideProps } from '@mui/types';
+import { SxProps } from '@mui/system';
+// @ts-ignore
+import Grid from '@pigment-css/react/Grid';
+import composeClasses from '@mui/utils/composeClasses';
+import generateUtilityClass from '@mui/utils/generateUtilityClass';
+import {
+  generateDirectionClasses,
+  generateSizeClassNames,
+  generateSpacingClassNames,
+} from '@mui/system/Unstable_Grid/gridGenerator';
+import { Breakpoint, Theme } from '../styles';
+import { useDefaultProps } from '../DefaultPropsProvider';
+
+type ResponsiveStyleValue<T> = T | Array<T | null> | { [key in Breakpoint]?: T | null };
+
+export type GridDirection = 'row' | 'row-reverse' | 'column' | 'column-reverse';
+
+export type GridSpacing = number | string;
+
+export type GridWrap = 'nowrap' | 'wrap' | 'wrap-reverse';
+
+export type GridSize = 'auto' | 'grow' | number;
+
+export interface GridBaseProps {
```

Comment:
Maybe not for this PR, but should we consider sharing types between the components and their pigment version? I think this would help us keep the interchangeable